packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -325,6 +412,141 @@ export class ModelConnectionTransformer extends Transformer {
+     * The directive is given an index to query and a list of fields to query by such that it
+     * returns a list objects (or in certain cases a single object) that are connected to the
+     * object it is called on.
+     * This directive is designed to leverage indices configured using @key to create relationships.
+     *
+     * Directive Definition:
+     * @connection(keyName: String, fields: [String!]!) on FIELD_DEFINITION
+     * param @keyName The name of the index configured using @key that should be queried to get
+     *      connected objects
+     * param @fields The names of the fields on the current object to query by.
+     */
+    public newParameterization = (
+        parent: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode,
+        field: FieldDefinitionNode,
+        directive: DirectiveNode,
+        ctx: TransformerContext
+    ): void => {
+        const parentTypeName = parent.name.value;
+        const fieldName = field.name.value;
+        const args : RelationArguments = getDirectiveArguments(directive);
+        const numFields = args.fields.length;
+
+        // Check that related type exists and that the connected object is annotated with @model.
+        const relatedTypeName = getBaseType(field.type)
+        const relatedType = ctx.inputDocument.definitions.find(
+            d => d.kind === Kind.OBJECT_TYPE_DEFINITION && d.name.value === relatedTypeName
+        ) as ObjectTypeDefinitionNode | undefined
+
+        // Get Child object's table.
+        const tableLogicalID = ModelResourceIDs.ModelTableResourceID(relatedType.name.value);
+        const tableResource = ctx.getResource(tableLogicalID) as Table;
+
+        // Ensure that there is at least one field provided.
+        if (numFields === 0) {
+            throw new InvalidDirectiveError('No fields passed in to @connection directive.')
+        }
+
+        // Check that each field provided exists in the parent model and that it is a valid key type (single non-null).
+        let inputFields : FieldDefinitionNode[] = [];
+        args.fields.forEach(
+            item => {
+                const fieldsArrayLength = inputFields.length;
+                inputFields[fieldsArrayLength] = parent.fields.find(f => f.name.value === item);
+                if (!inputFields[fieldsArrayLength]) {
+                    throw new InvalidDirectiveError(item + ' is not a field in ' + parentTypeName)
+                }
+
+                validateKeyFieldNewConnection(inputFields[fieldsArrayLength]);
+            })
+
+        let index : GlobalSecondaryIndex = undefined;
+        // If no index is provided use the default index for the related model type and
+        // check that the query fields match the PK/SK of the table. Else confirm that index exists.
+        if (!args.keyName) {
+            checkFieldsAgainstIndex(parent.fields, relatedType.fields, args.fields, tableResource.Properties.KeySchema);
```

Comment:
If we've a ```keyName``` specified, then shouldn't we validate that the a ```@key``` exists on the connected type?